### PR TITLE
Srst2 fixes

### PR DESCRIPTION
**Pull request by markiskander**
Made srst2 not fail when used with data with spaces in the name. Also fixed linting and added some acknowledgements.

### CODE REVIEW

**Comment by Takadonet** [comment]
galaxy_tools/packages/package_srst2_0_1_4_5/ should only contain the tool_dependencies.xml

**Comment by Takadonet** [comment]
getmlst.py should be it's own tool under tools/ folder. 
package_srst2_0_1_4_5 is the binary for both getmlst.py and srst2.py

File: tools/srst2/srst2.pl [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+
+    symlink($ARGV[$index_offset], $database);
+    $ARGV[$index_offset] = $database;
+
+    $index_offset++;
+  }
+}
+
+#my $command = "python $binary @ARGV";
+my $command = "/usr/bin/python $binary @ARGV";
```
Comment (Takadonet) [inline tools/srst2/srst2.pl]: delete ln 137 and uncomment 136.

File: tools/srst2/srst2.xml [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+Original Author: Mariam Iskander
+
+Jen Cabral
+Philip Mabon
+Mark Iskander
+
+    </help>
+
+    <citations>
+    </citations>
```
Comment (Takadonet) [inline tools/srst2/srst2.xml]: add the DOI citation of 10.1128/AAC.01310-13

Example on how to use it @ https://wiki.galaxyproject.org/Admin/Tools/ToolConfigSyntax#A.3Ccitations.3E_tag_set

File: tools/srst2/srst2.pl [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+# concatenate all pileup files
+if (@pileupfiles > 1){
+  $cmd = "cat ".join(" ",@pileupfiles)." > $pileup";
+  system($cmd);
+} else {
+  move($pileupfiles[0], $pileup);
+}
+
+# perform find-replace to restore original user-specified file names
+for (my $i= 0; $i < @genefiles; $i++){
```
Comment (Takadonet) [inline tools/srst2/srst2.pl]: Use a foreach here instead. i.e

foreach my $gene(@genefiles) {

File: tools/srst2/srst2.pl [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+  my $data = read_file($genefiles[$i]);
+  $data =~ s/temp_file_name/$file_name[0]/g;
+  write_file($genefiles[$i], $data);
+}
+
+for (my $i= 0; $i < @fullgenefiles; $i++){
```
Comment (Takadonet) [inline tools/srst2/srst2.pl]: use foreach please

**Comment by markiskander** [comment]
Alright, I think I got everything sorted but let me know if something's off

**Comment by Takadonet** [comment]
Looks good.

**Pull request merged**